closed/src/java.base/share/classes/sun/security/provider/NativeDigest.java

```diff
@@ -62,88 +57,24 @@
     private long bytesProcessed;
 
     private static NativeCrypto nativeCrypto;
-    private NativeCryptoContextManager ctxMgr;
-
-    private static volatile NativeCryptoContextManager cryptoContextMgrSingleton = null;
-
-    public NativeCryptoContextManager getNativeCryptoContextManagerInstance() {
-          if (cryptoContextMgrSingleton == null) {
-              synchronized (NativeCryptoContextManager.class) {
-                if (cryptoContextMgrSingleton == null) {
-                    cryptoContextMgrSingleton = new NativeCryptoContextManager();
-                }
-              }
-          }
-          return cryptoContextMgrSingleton;
-    }
+    private static Cleaner digestCleaner;
```

Comment:
This field and `nativeCrypto` should both be `final`.

Reply:
updated